app/controllers/concerns/handle_empty_email.rb

```diff
@@ -1,16 +1,40 @@
-
+# This module overrides Blacklight's mail-sending methods to only
+# actually send an email if there are documents selected.
+# MK 9/2020 - Why? Dunno. Maybe crawlers hit the urls and this was throwing
+# exceptions?
 module HandleEmptyEmail
 
   extend ActiveSupport::Concern
 
-  # override
+  # override of method in Blacklight::Catalog
   def email_action(documents)
-      super(documents) if documents.length > 0
+    return unless documents.any? && email_is_legit(params[:to])
+
+    begin
+      retries ||= 0
+      super(documents)
+    rescue Net::ReadTimeout => e
+      sleep 3
+      if (retries += 1) < 2
+        retry
+      else
+        Honeybadger.notify e
```

Comment:
I would just re-raise the error, that way if we ever switch away from Honeybadger we don't have to change this reference.

Reply:
done